src/transformers/models/deformable_detr/modeling_deformable_detr.py

```diff
@@ -1417,6 +1438,18 @@ def get_proposal_pos_embed(self, proposals):
         return pos
 
     def gen_encoder_output_proposals(self, memory, memory_padding_mask, spatial_shapes):
+        """Generate the encoder output proposals from encoded memory.
+
+        Args:
+            memory (Tensor[batch_size, num_reference_points, d_model]): Output of the encoder.
+            memory_padding_mask (Tensor[batch_size, num_reference_points]): Padding mask for memory.
+            spatial_shapes (Tensor[num_feature_levels, 2]): Spatial shapes of the feature maps.
```

Comment:
```suggestion
            memory (torch.FloatTensor of shape `(batch_size, num_reference_points, d_model)`): Output of the encoder.
            memory_padding_mask (torch.LongTensor of shape `(batch_size, num_reference_points)`): Padding mask for memory.
            spatial_shapes (torch.LongTensor of shape `(num_feature_levels, 2)`): Spatial shapes of the feature maps.
```
```suggestion
            memory (Tensor[batch_size, num_reference_points, d_model]): Output of the encoder.
            memory_padding_mask (Tensor[batch_size, num_reference_points]): Padding mask for memory.
            spatial_shapes (Tensor[num_feature_levels, 2]): Spatial shapes of the feature maps.
```
I guess the first one is a float tensor and the other two long tensors, correct me if I'm wrong